packages/service-provider-server/src/cli-service-provider.ts

```diff
@@ -1208,7 +1210,15 @@ class CliServiceProvider extends ServiceProviderCore implements ServiceProvider
     });
   }
 
-  // TODO(MONGOSH-1456): getSearchIndexes
+  listSearchIndexes(
+    database: string,
+    collection: string,
+    options: ListSearchIndexesOptions,
+    dbOptions?: DbOptions): ListSearchIndexesCursor {
+    return this.db(database, dbOptions)
+      .collection(collection)
+      .listSearchIndexes(options);
+  }
```

Comment:
If we’re not exposing `listSearchIndexes`, I’d probably also align this with `getIndexes` here in the SP interface. Passing an array across the interface boundary is probably easier to implement in alternative SP implementations like the java-shell one than passing a (new type of?) Cursor.

Reply:
I just thought we'd ultimately end up with a larger interface? I don't feel too strongly about it, though